File: dispatch/java/runtime/adamma/c4dhi/claid/Module/Scheduling/ScheduledRunnable.java
package adamma.c4dhi.claid.Module.Scheduling;

public abstract class ScheduledRunnable
{
    protected ScheduleDescription schedule;
    protected boolean valid = false;

    public ScheduledRunnable(ScheduleDescription schedule)
    {
        this.schedule = schedule;
        this.valid = true;
    }

    public abstract void run();

    public ScheduleDescription getSchedule()
    {
        return this.schedule;
    }

    public boolean isValid()
    {
        return this.valid;
    }

    public void invalidate()
    {
        this.valid = false;
    }
    
}

File: dispatch/ios/CLAID/Sources/CLAID/Module/Scheduling/ScheduledRunnable.swift
//
//  ScheduledRunnable.swift
//  CLAID
//
//  Created by Patrick Langer on 19.02.2025.
//

public protocol ScheduledRunnable: Sendable {
    var schedule: ScheduleDescription { get }
    func isValid() async -> Bool
    
    func run() async
    func invalidate() async
    
}
